huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_pair.go

```diff
@@ -0,0 +1,212 @@
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	n, err := replications.Get(client, d.Id()).Extract()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.1608" {
+				// `SDRS.1608` means replication pair not found
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error retrieving SDRS replication pair")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	volumes := strings.Split(n.VolumeIDs, ",")
+	if len(volumes) != 2 {
+		return diag.Errorf("error retrieving volumes of replication pair: Invalid format. "+
+			"except retrieving 2 volumes, but got %d", len(volumes))
+	}
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("name", n.Name),
+		d.Set("group_id", n.GroupID),
+		d.Set("volume_id", volumes[0]),
+		d.Set("description", n.Description),
+		d.Set("replication_model", n.ReplicaModel),
+		d.Set("fault_level", n.FaultLevel),
+		d.Set("status", n.Status),
+		d.Set("target_volume_id", volumes[1]),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceReplicationPairUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	if d.HasChange("name") {
+		updateOpts := replications.UpdateOpts{
+			Name: d.Get("name").(string),
+		}
+		_, err = replications.Update(client, d.Id(), updateOpts).Extract()
+		if err != nil {
+			return diag.Errorf("error updating SDRS replication pair, %s", err)
+		}
+	}
+	return resourceReplicationPairRead(ctx, d, meta)
+}
+
+func resourceReplicationPairDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	deleteOpts := replications.DeleteOpts{
+		GroupID:      d.Get("group_id").(string),
+		DeleteVolume: d.Get("delete_target_volume").(bool),
+	}
+	n, err := replications.Delete(client, d.Id(), deleteOpts).ExtractJobResponse()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.1608" {
+				// `SDRS.1608` means replication pair not found
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error deleting SDRS replication pair")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	deleteTimeoutSec := int(d.Timeout(schema.TimeoutDelete).Seconds())
+	if err := replications.WaitForJobSuccess(client, deleteTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
```

Comment:
```suggestion
	d.SetId("")
	return nil
```

Reply:
The deletion method will automatically set the ID to empty internally.

Reply:
ok